// wdio-vscode-service/dist/pageobjects/sidebar/tree/default/DefaultTreeItem.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var DefaultTreeItem_1;
import { TreeItem } from '../../ViewItem.js';
import { PageDecorator } from '../../../utils.js';
import { TreeItem as TreeItemLocators, DefaultTreeItem as DefaultTreeItemLocators } from '../../../../locators/1.73.0.js';
/**
 * Default tree item base on the items in explorer view
 *
 * @category Sidebar
 */
let DefaultTreeItem = DefaultTreeItem_1 = class DefaultTreeItem extends TreeItem {
    constructor(locators, element, viewPart) {
        super(locators, element, viewPart.elem);
        this.viewPart = viewPart;
        /**
         * @private
         */
        this.locatorKey = ['TreeItem', 'DefaultTreeItem'];
    }
    getDescription() {
        return Promise.resolve(undefined);
    }
    async getLabel() {
        return this.elem.getAttribute(this.locatorMap.DefaultTreeSection.itemLabel);
    }
    async getTooltip() {
        return this.tooltip$.getAttribute('title');
    }
    async isExpanded() {
        const twistieClass = await this.twistie$.getAttribute('class');
        return twistieClass.indexOf('collapsed') < 0;
    }
    async getChildren() {
        const rows = await this.getChildItems(this.locatorMap.DefaultTreeSection.itemRow);
        const items = await Promise.all(rows.map(async (row) => (new DefaultTreeItem_1(this.locatorMap, 
        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
        row, this.viewPart).wait())));
        return items;
    }
    async isExpandable() {
        const twistieClass = await this.twistie$.getAttribute('class');
        return twistieClass.indexOf('collapsible') > -1;
    }
};
DefaultTreeItem = DefaultTreeItem_1 = __decorate([
    PageDecorator({ ...TreeItemLocators, ...DefaultTreeItemLocators })
], DefaultTreeItem);
export { DefaultTreeItem };
